Library/Homebrew/cmd/update-bash.sh

```diff
@@ -2,22 +2,41 @@ brew() {
   "$HOMEBREW_BREW_FILE" "$@"
 }
 
+git() {
+  [[ -n "$HOMEBREW_GIT" ]] || exit 1
+  "$HOMEBREW_GIT" "$@"
+}
+
 which_git() {
-  local which_git
+  local git_path
   local active_developer_dir
 
-  which_git="$(which git 2>/dev/null)"
-  if [[ -n "$which_git" && "/usr/bin/git" = "$which_git" ]]
+  if [[ -n "$HOMEBREW_GIT" && -x "$HOMEBREW_GIT" ]]
+  then
+    git_path="$HOMEBREW_GIT"
+  elif [[ -n "$GIT" && -x "$GIT" ]]
+  then
+    git_path="$GIT"
+  else
+    git_path="$(which git 2>/dev/null)"
+  fi
+
+  if [[ -n "$git_path" ]]
+  then
+    git_path="$(chdir "${git_path%/*}" && pwd -P)/${git_path##*/}"
```

Comment:
> OK, I will change it to `dirname` when pulling.

That's not what I was trying to imply. Does it make sense to have something like `git-2.2` in one of the variables, with the expectation that it will be looked up in `PATH`? If yes, the whole construct doesn't make sense and will fail, no matter if we use `dirname` or not.